frontend/src/metabase/visualizations/echarts/cartesian/scatter/series.ts

```diff
@@ -1,19 +1,50 @@
+import d3 from "d3";
 import type { RegisteredSeriesOption } from "echarts/types/dist/shared";
 
 import type { RenderingContext } from "metabase/visualizations/types";
 
-import type { SeriesModel } from "../model/types";
+import type {
+  DataKey,
+  Datum,
+  GroupedDataset,
+  SeriesModel,
+} from "../model/types";
+
+const BUBBLE_SCALE_FACTOR_MAX = 64;
+
+// TODO: refine the scaling curve when implementing the dynamic scatter plot
+function getBubbleSizeScale(
+  dataset: GroupedDataset,
+  bubbleSizeDataKey: DataKey | undefined,
+) {
+  if (!bubbleSizeDataKey) {
+    return 8;
+  }
+
+  const bubbleSizeDomainMax = d3.max(
```

Comment:
We compute series extents in the model so it would be nice to reuse it

Reply:
Done, I've also added another storybook example to test the right y-axis since that changes based on the yAxisExtents

Reply:
Should it be the extend of the bubble size series? If you look at `yAxisExtents` then it will be the extent of a metric that is used for y-axis.
At the model level we compute extents for series by dataKeys like `const extents = getDatasetExtents(seriesDataKeys, dataset)` and I believe we need to compute it for the bubble size column and then use it here

Reply:
Ah you're right, sorry I didn't realize we also computed the extents for each `dataKey`. I'll fix this

Reply:
@alxnddr I looked into it and we don't actually compute the extents for the bubble size column. `getDatasetExtents` is called with `seriesDataKeys`, but there is no series data key for bubble size, because there is no series model for it. We don't have a series model for it because bubble size is not a series on the chart, e.g. we don't render a separate series of bubbles for it using `buildEChartsSeries`, which calls `buildEChartsScatterSeries` on each series model.

I think we should just revert to calculating the extent here, since it was not calculated with `getDatasetExtents` and it doesn't make sense to add a separate series for it.